testing/board/test_board_vc.cpp

```diff
@@ -0,0 +1,358 @@
+
+		filter_ = nullptr;
+		ASSERT_EQ(fpgaInitialize(NULL), FPGA_OK);
+		ASSERT_EQ(fpgaGetProperties(nullptr, &filter_), FPGA_OK);
+		ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_DEVICE), FPGA_OK);
+		num_matches_ = 0;
+		ASSERT_EQ(fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(),
+			&num_matches_), FPGA_OK);
+		EXPECT_GT(num_matches_, 0);
+		dev_ = nullptr;
+		ASSERT_EQ(fpgaOpen(tokens_[0], &dev_, 0), FPGA_OK);
+	}
+
+	virtual void TearDown() override {
+		EXPECT_EQ(fpgaDestroyProperties(&filter_), FPGA_OK);
+		if (dev_) {
+			EXPECT_EQ(fpgaClose(dev_), FPGA_OK);
+			dev_ = nullptr;
+		}
+		for (auto &t : tokens_) {
+			if (t) {
+				EXPECT_EQ(fpgaDestroyToken(&t), FPGA_OK);
+				t = nullptr;
+			}
+		}
+		fpgaFinalize();
+		system_->finalize();
+	}
+
+	std::array<fpga_token, 2> tokens_;
+	fpga_properties filter_;
+	fpga_handle dev_;
+	test_platform platform_;
+	uint32_t num_matches_;
+	test_system *system_;
+};
+
+ssize_t board_vc_c_p::eintr_write(int fd, void *buf, size_t count)
+{
+	ssize_t bytes_written = 0, total_written = 0;
+	char *ptr = (char*)buf;
+
+	if (!buf) {
+		return -1;
+	}
+
+	while (total_written < (ssize_t)count) {
+		bytes_written =
+			write(fd, ptr + total_written, count - total_written);
+		if (bytes_written < 0) {
+			if (errno == EINTR) {
+				continue;
+			}
+			return bytes_written;
+		}
+		total_written += bytes_written;
+	}
+	return total_written;
+
+}
+fpga_result board_vc_c_p::write_sysfs_file(const char *file,
+	void *buf, size_t count) {
+	fpga_result res = FPGA_OK;
+	char sysfspath[SYSFS_MAX_SIZE];
+	int fd = 0;
+
+	snprintf_s_ss(sysfspath, sizeof(sysfspath), "%s/%s", "/sys/class/fpga/intel-fpga-dev.0/intel-fpga-fme.0", file);
+
+	glob_t pglob;
+	int gres = glob(sysfspath, GLOB_NOSORT, NULL, &pglob);
+	if ((gres) || (1 != pglob.gl_pathc)) {
+		globfree(&pglob);
+		return FPGA_NOT_FOUND;
+	}
+	fd = open(pglob.gl_pathv[0], O_WRONLY);
+	globfree(&pglob);
+	if (fd < 0) {
+		printf("open faild \n");
+		return FPGA_NOT_FOUND;
+	}
+
+	ssize_t total_written = eintr_write(fd, buf, count);
+	if (total_written == 0) {
+		close(fd);
+		printf("total_written failed \n");
+		return FPGA_INVALID_PARAM;
+	}
+
+	close(fd);
+	return res;
+}
+
+fpga_result board_vc_c_p::delete_sysfs_file(const char *file) {
+	fpga_result res = FPGA_OK;
+	char sysfspath[SYSFS_MAX_SIZE];
+	int status = 0;
+
+	snprintf_s_ss(sysfspath, sizeof(sysfspath), "%s/%s", "/sys/class/fpga/intel-fpga-dev.0/intel-fpga-fme.0", file);
+
+	glob_t pglob;
+	int gres = glob(sysfspath, GLOB_NOSORT, NULL, &pglob);
+	if ((gres) || (1 != pglob.gl_pathc)) {
+		globfree(&pglob);
+		return FPGA_NOT_FOUND;
+	}
+	status = remove(pglob.gl_pathv[0]);
+
+	globfree(&pglob);
+	if (status < 0) {
+		printf("delete failed = %d \n", status);
+		return FPGA_NOT_FOUND;
+	}
+
+	return res;
+}
+
+/**
+* @test       board_vc_1
+* @brief      Tests: read_bmcfw_version
+* @details    Validates bmc firmware version  <br>
+*/
+TEST_P(board_vc_c_p, board_vc_1) {
```

Comment:
Should we also have a test that verifies that the value returned by read_bmcfw_version() is correct? Same for all the other tests here that read in a value?